homeassistant/components/roborock/services.yaml

```diff
@@ -26,3 +26,21 @@ get_vacuum_current_position:
     entity:
       integration: roborock
       domain: vacuum
+vacuum_clean_rooms:
+  target:
+    entity:
+      integration: roborock
+      domain: vacuum
+  fields:
+    room_ids:
+      example: 1,3,4
+      required: true
+      selector:
+        text:
+    repeat:
+      example: 1
+      required: false
+      selector:
+        number:
+          min: 1
+          max: 3
```

Comment:
@allenporter There isn't a way that this could be dynamic is there? We have individual room names and that would be a lot better for the user, but if I'm not mistaken it isn't really possible to do dynamic fields in Actions.

Reply:
I've already asked myself the question, due to the usability, the potential usability in Homekit/Matter (I think vaccums are supported from iOS 18.3) and the usability in HA Voice Assistant (which I haven't looked into yet) In a perfect world, this would also correspond to the rooms in HomeAssistant

Reply:
I believe I tried this in the past and I came up short and decided I would wait for [Multiselect entities](https://github.com/home-assistant/architecture/discussions/927) but they are indefinitely on hold now.